### DISCUSSION TITLE
Implement Windows taskbar extensions such as right-clicking to view recent files

### DISCUSSION THREAD

**Discussion by mika76** [godotengine/godot-proposals#9085] [Platforms]
### Describe the project you are working on

Any project - this is more about working with and within Godot on Windows

### Describe the problem or limitation you are having in your project

Godot is missing some windows explorer/taskbard features which have become pretty standard across apps. Recent files/jump lists on taskbar icons, progress on taskbar icons, icons for Godot specific file extensions, thumbnails for specific files (maybe scene files), etc... - pretty much all the stuff here: https://learn.microsoft.com/en-us/windows/win32/shell/taskbar-extensions

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Implement them. Recent files would be the top of the priority list - it would be so much easier to be able to pop into a project immediately by selecting a recent project from the taskbar and not by having to open to Godot's splash screen first.

Progress would probably be priority 2 since it's quite useful to watch it working and see when it's done while an app is minimized for example.

The rest are all nice-to-haves but would up the quality of Godot on Windows.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

N/A

### If this enhancement will not be used often, can it be worked around with a few lines of script?

N/A

### Is there a reason why this should be core and not an add-on in the asset library?

I have no idea if any of these could be done as addons...

**Comment by mika76** [comment]
PS: I see some of the stuff was discussed in [#5653](https://github.com/godotengine/godot/issues/5653) but it was closed because feature requests were being moved here - but then nobody seems to have re-created it.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Comment by gaudimancer-creator** [comment]
Hello, 
Is this still in discussion ? The "Thumbnails Toolbar" (https://learn.microsoft.com/en-us/windows/win32/shell/taskbar-extensions#thumbnail-toolbars) would be useful for a lot of people.

**Reply by Calinou** [reply]
To my knowledge, nobody is currently working on implementing this.

Note that KDE has some equivalents for this on Linux, at least for music playback (although this might tie to [existing OS media playback APIs](https://github.com/godotengine/godot-proposals/issues/8582)). macOS' Dock also supports defining app-specific actions on right-click.
